dusk-network/dusk-blockchain
# add utils from master to rusk-ipc branch and fix issues

* add utils from master to rusk-ipc branch and fix issues
* add Makefile task for keepalive version of test harness
* Fix RPC connection issues
* FIX GQL query issues
* update harness test to use rusk mock and grpc mocks as separated processes that can keep alive after test is finished
* add new cmd option on utils for mockCMD and mockRUSKCMD
* add task for mock and mockrusk to Makefile
* update test-harness-ci and test-harness-alive in order to allow for the new configs and add stop to kill utils

## How to test
1. Start test harness with keep alive
```
make test-harness-alive
```

2. Start metrics collection:
```
make build && ./bin/utils metrics
```

3. Hit the metrics endpoint and observe transactions being generated and stats being collected:
```
watch "curl http://localhost:9099/metrics"
```

4. Observe the blocks are building non stop
```
dusk_block 6
dusk_seconds_last_block 793.23
dusk_effective_block_time 0
dusk_block_transactions 0
dusk_pending_transactions 0
dusk_block_tps 0
dusk_block_value 0
dusk_block_size_bytes 0
dusk_contracts_created 0
dusk_token_transfers 0
dusk_transfers 0
dusk_load_time 0.0000
```

## New Utils CMD for mock processes

1. gRPC mock for RuskMock, WalletMock, TransactorMock
How to use: 
```
make build && ./bin/utils mock --grpcmockhost=127.0.0.1:9191
```

2. gRPC RUSK mock:
```
make build && ./bin/utils mockrusk --rusknetwork=tcp --ruskaddress=127.0.0.1:10000 \
	--walletstore=/tmp/localnet-137601832/node-9003/walletDB/ \
	--walletfile=./harness/data/wallet-9000.dat
```
Note: Make sure to update the `walletstore` to the appropriate directory where the wallet db in question is at.

closes  #556
